src/components/CardDetail/CardDetail.tsx

```diff
@@ -0,0 +1,240 @@
+import type { BasePokemon, PokemonSpecies } from '@types';
+import { useEffect, useState } from 'react';
+import './index.css';
+import { fetchPokemonBySpecies } from '@api/fetchPokemonBySpecies';
+import Spinner from '@components/Spinner';
+
+interface CardDetailProps {
+  pokemon: BasePokemon;
+  onClose: () => void;
+}
+
+const CardDetail = ({ pokemon, onClose }: CardDetailProps) => {
+  const [imageError, setImageError] = useState(false);
+  const [loading, setLoading] = useState(true);
```

Comment:
Булевские значения лучше именовать так
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```